apps/publications/src/app/dialogs/add-thanks-dialog/add-thanks-dialog.component.ts

```diff
@@ -27,37 +33,29 @@ export class AddThanksDialogComponent implements OnInit {
     private cabinetManagerService: CabinetManagerService
   ) {}
 
-  loading: boolean;
-  owners: Owner[];
-  selected = new SelectionModel<Owner>(true, []);
-  filterValue: string;
-  tableId = TABLE_ADD_THANKS_DIALOG;
-
   ngOnInit(): void {
     this.loading = true;
-    const allowedOwners = this.storeService.get('allowed_owners_for_thanks');
+    const allowedOwners = this.storeService.get('allowed_owners_for_thanks') as number[];
     this.ownersManagerService.getAllOwners().subscribe((owners) => {
       if (allowedOwners.length !== 0) {
-        this.owners = owners.filter((item) => allowedOwners.indexOf(item.id) > -1);
+        this.owners = owners.filter((item) => allowedOwners.includes(item.id));
```

Comment:
I dont think this filtering works, even if I insert something into the allowed_owners_for_thanks, there is no owner I can add
also I wanted to be the default value - when the field is empty - all owners can be thanked

Reply:
fixed filtering